File: app/src/app/offerRide.ts

export class OfferRide{
    startpoint: string;
    endPoint: string;
    date:string;
    timeSlot:string;
    seats:number;
    price:string;
    stops:string;
    userName:string;
    userId:number;
    
    constructor(startpoint: string,endPoint: string, date:string ,timeSlot:string, seats:number, stops:string, price:string, userName:string,userId:number
        ){
        this.startpoint=startpoint;
        this.endPoint = endPoint;
        this.date= date;
        this.timeSlot = timeSlot;
        this.seats= seats;
        this.price = price;
        this.userName = userName;
        this.userId = userId;
        this.stops =stops;
    }
}

File: Data/Models/OfferRide.cs
using System.ComponentModel.DataAnnotations;

namespace CarPooling.Data.Models
{
    public class OfferRide
    {
        [Key]
        public Guid OfferRideId { get; set; }
        public int UserId { get; set; }
        public string? UserName { get; set; }
        public string StartPoint { get; set; } = null!;
        public string? stops { get; set; } 
        public string EndPoint { get; set; } = null!;
        public string Date { get; set; } = null!;
        public string TimeSlot { get; set; } = null!;
        public int Seats { get; set; } = 0;
        public string Price { get; set; } = null!;
    }
}

